independent-projects/tools/platform-descriptor-resolver-json/src/main/java/io/quarkus/platform/descriptor/resolver/json/QuarkusJsonPlatformDescriptorResolver.java

```diff
@@ -58,21 +59,43 @@
     public static final String PROP_PLATFORM_JSON_VERSION = "quarkus.platform.json.version";
     public static final String PROP_PLATFORM_JSON_VERSION_RANGE = "quarkus.platform.json.version-range";
 
+    private static final JsonDescriptorResolver jsonDescriptorResolver;
+    static {
+        jsonDescriptorResolver = new ChainedJsonDescriptorResolver(
+                ((bomGroupId, bomArtifactId, bomVersion, jsonResolver, log) -> jsonResolver.resolveArtifact(bomGroupId,
+                        bomArtifactId + "-" + BootstrapConstants.PLATFORM_DESCRIPTOR_ARTIFACT_ID_SUFFIX, bomVersion, "json",
+                        bomVersion, log)),
+                ((bomGroupId, bomArtifactId, bomVersion, jsonResolver, log) -> jsonResolver.resolveArtifact(bomGroupId,
+                        bomArtifactId, null, "json", bomVersion, log)),
+                ((bomGroupId, bomArtifactId, bomVersion, jsonResolver, log) -> jsonResolver.resolveArtifact(bomGroupId,
+                        bomArtifactId + "-descriptor-json", null, "json", bomVersion, log)));
+    }
+
+    public static JsonDescriptorResolver jsonDescriptorResolver() {
+        return jsonDescriptorResolver;
+    }
```

Comment:
@gastaldi this PR changes the coordinates of the JSON descriptor and will require adjustments in the extension catalog project. To hopefully make it easier to maintain, I introduced this method and a couple interfaces right above this class. This specific method returns a JSON resolver that will try resolving the JSON for the BOM coordinates using the legacy and the new JSON coordinates conventions.

Reply:
@aloubyansky No problem, it's just a matter of changing the `platforms.json` to the new GAV